Documentation/quickstart/debugging.rst

```diff
@@ -93,11 +98,11 @@ You can add your own custom debug logging by adding the following lines to
    #  define custinfo    _none
    #endif
 
-You need to add the following line to your ``.config`` file:
+You need to enable the setting this way:
 
-.. code-block:: c
+.. code-block:: console
 
-   CONFIG_DEBUG_CUSTOM_INFO=y
+   $ kconfig-tweak --enable CONFIG_CUSTOM_INFO
```

Comment:
I dont think this will work because there is no Kconfig that defines CUSTOM_INFO.  Might be best in the block above to not depend on CONFIG_DEBUG_CUSTOM_INFO and just do the `#define custinfo  _info`.  Or recommend just using `_info` for debugging.

Reply:
@btashton using _info for logging produces too much output... How about I add a kconfig entry for CUSTOM_INFO? I could also add the code to debug.h too. That would save people from having to add it.

It's super handy to have a logging statement that you can use for your own purposes that no other module is using.

Reply:
I would be open to that with the clear expectation that we would not allow usage of it to be merged in.

Reply:
also if you do add that, the `custinfo` should be defined in the header file

Reply:
@btashton Ok, I added custinfo to debug.h, CUSTOM_INFO to the Kconfig file, and updated docs.

I see what you mean about using the Kconfig menu system to set this– testing it, I can see how it work. Some good instructions would be helpful as an alternative to the kconfig-tweak stuff... but it really needs an asciinema movie to show people what to do... maybe in another PR??

Reply:
@btashton I added a comment in debug.h that says "do not create PRs that include usages of custinfo" – let me know if you want a different message there.

Reply:
Where is this going to be used?  If that's something just to track an issue, why not just `printf` that we delete afterwards?
If this is part of a driver, I think using what's available* is readable enough. 

*: I mean ainfo for ADC, lcdinfo for LCDs, etc.

Reply:
@Ouss4 I didn't find `printf` was enough– I wanted the timestamps that the logging facility produces, so would have had to reinvent that myself if I used `printf` – and `printf` also cannot use the in-memory logging.

While porting the imxrt SDMMC driver to sama5, I found it very helpful to have an independent logging facility that I could use to track timing issues across multiple modules and multiple threads, without unleashing 🙂 a torrent of logging from other modules.

Reply:
@adamfeuer this is where I am also confused because developing that driver I would expect you would have used `mcinfo` which is controlled by `CONFIG_DEBUG_MEMCARD_INFO`  and Kconfig describes as "Enable MMC/SD memory card driver informational output to SYSLOG."

example:  `arch/arm/src/stm32/stm32_sdio.c`

I guess I would just expect people to use the subsystem logger they are building on or add one if there is not an appropriate.

Reply:
I am going AFK for the rest of the holiday, so don't let me hold this up :)

Reply:
@btashton The issue with solely using `mcinfo` is that I had to look into several other different subsystems – `fs` which uses `fsinfo`, and a couple more that I don't remember right now. With a cross-cutting concern, if you turn on all the logging for several subsytems, that unleashes a torrent of logging. I needed something that I could add to different subsystems without turning everything on.

Anyway, if this is too specialized, I can either go back to having it be a tip in the docs, or remove it completely from the docs and just document it on the wiki or my own website.

Have a great holiday!

Reply:
@Ouss4 @btashton I removed the custinfo Kconfig, debug.h, and docs so that the rest of the error fixes can go in.